google-assistant-sdk/googlesamples/assistant/grpc/pushtotalk.py

```diff
@@ -133,11 +133,11 @@ def iter_assist_requests():
             assistant_helpers.log_assist_response_without_audio(resp)
             if resp.event_type == END_OF_UTTERANCE:
                 logging.info('End of audio request detected')
-                self.conversation_stream.stop_recording()
             if resp.speech_results:
                 logging.info('Transcript of user request: "%s".',
                              ' '.join(r.transcript
                                       for r in resp.speech_results))
+                self.conversation_stream.start_playback()
```

Comment:
Should playback start when we get text speech results? I think it makes sense to do it when we get audio data.

Reply:
Do you think to start playback in 'if len(resp.audio_out.audio_data) > 0:' ?  
Yes, it might be better way...

Reply:
Yes.

Reply:
Yes, but we also don't want to call `start_playback` more than one time.

Reply:
We should be able to check if `start_playback` has started already. Right now it feels like it is in the wrong place.

Reply:
@Fleker do you want me to change this in this CL?

Reply:
we should probably change it in this PR, but I'll leave it up to you.

Reply:
Done.